go-controller/pkg/node/gateway_iptables.go

```diff
@@ -61,10 +61,13 @@ func delIptRules(rules []iptRule) error {
 	for _, r := range rules {
 		klog.V(5).Infof("Deleting rule in table: %s, chain: %s with args: \"%s\" for protocol: %v ", r.table, r.chain, strings.Join(r.args, " "), r.protocol)
 		ipt, _ := util.GetIPTablesHelper(r.protocol)
-		err := ipt.Delete(r.table, r.chain, r.args...)
-		if err != nil {
-			return fmt.Errorf("failed to delete iptables %s/%s rule %q: %v",
-				r.table, r.chain, strings.Join(r.args, " "), err)
+		if exists, err := ipt.Exists(r.table, r.chain, r.args...); err == nil && exists {
+			err := ipt.Delete(r.table, r.chain, r.args...)
+			if err != nil {
+				return fmt.Errorf("failed to delete iptables %s/%s rule %q: %v",
```

Comment:
do we really want to return if we cant delete one rule? Should we continue trying to remove all of the rules?

Reply:
Fixed it to use `errors.Wrapf` like we do in other places around the code.